addons/account_edi/models/account_journal.py

```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, fields
+
+
+class AccountJournal(models.Model):
+    _inherit = 'account.journal'
+
+    edi_ids = fields.Many2many(comodel_name='account.edi', string='EDI Format', help='When sending an invoice, the following XML will be attached or embedd to the PDF.')
+
+    def _create_invoice_from_attachment(self, attachment):
+        invoice = self.env['account.edi'].search([])._create_invoice_from_attachment(attachment)
+        if invoice:
+            invoice.with_context(no_new_invoice=True).message_post(attachment_ids=[attachment.id])
+            return invoice
+        else:
+            return super()._create_invoice_from_attachment(attachment)
```

Comment:
``` diff
-def _create_invoice_from_attachment(self, attachment):
-        invoice = self.env['account.edi'].search([])._create_invoice_from_attachment(attachment)
-        if invoice:
-            invoice.with_context(no_new_invoice=True).message_post(attachment_ids=[attachment.id])
-            return invoice
-        else:
-            return super()._create_invoice_from_attachment(attachment)
+def _create_invoice_from_attachment(self, attachment):
+        invoice = super()._create_invoice_from_attachment(attachment)
+        self.env['account.edi.format'].search([])._import_invoice_from_attachment(attachment, invoice=invoice)
+        return invoice
```

Reply:
This is wrong. Some xml could contains multiple invoices. So we need to make the distinction between updating and creating new invoices. In l10n_it, for example, we don't want to update an existing invoice if the xml contains multiple invoices but we allow importing the xml directly to create multiple invoices. Then, `self.env['account.edi'].search([])._create_invoice_from_attachment(attachment)` can return multiple invoices (I agree this is not very clear in the code).

Reply:
forget that